pkg/asset/manifests/cloudproviderconfig.go

```diff
@@ -253,6 +253,14 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 		vpcSubnets := installConfig.Config.PowerVS.VPCSubnets
 		if vpc == "" {
 			vpc = fmt.Sprintf("vpc-%s", clusterID.InfraID)
+		} else {
+			existingSubnets, err := installConfig.PowerVS.GetVPCSubnets(context.TODO(), vpc)
+			if err != nil {
+				return err
+			}
+			for _, subnet := range existingSubnets {
+				vpcSubnets = append(vpcSubnets, *subnet.Name)
+			}
```

Comment:
Aren't these subnets supplied via `installConfig.Config.PowerVS.VPCSubnets`?

Reply:
they are but we've found the provider will fail if you don't add all of the existing subnets. We could require that you specify all the subnets through the install config, but this is a more automated approach.

Reply:
Ok. Let's add that explanation as a comment here?

Reply:
will do!